app/forms/flood_risk_engine/steps/local_authority_address_form.rb

```diff
@@ -0,0 +1,49 @@
+module FloodRiskEngine
+  module Steps
+
+    class LocalAuthorityAddressForm < BaseAddressForm
+
+      def self.factory(enrollment)
+        raise(FormObjectError, "No Organisation set for step #{enrollment.current_step}") unless enrollment.organisation
+
+        new(FloodRiskEngine::Address.new(address_type: :primary), enrollment)
+      end
+
+      def prepopulator(options = {})
+        super(options)
+
+        # This is first stab at ensuring we can handle Back button/link
+        if @post_code.blank?
+          existing_address = enrollment.organisation.primary_address
```

Comment:
I'd suggest in `enrollment` delegating `primary_address` to `organisation` as this would greatly simplify this bit of code - that it, you wouldn't need to do so many checks that objects in the chain exist.